src/native/libs/System.Native/pal_uid.c

```diff
@@ -237,3 +237,55 @@ int32_t SystemNative_GetGroups(int32_t ngroups, uint32_t* groups)
 
     return getgroups(ngroups, groups);
 }
+
+int32_t SystemNative_GetUName(uint32_t uid, char* buffer, size_t bufferSize)
+{
+    assert(buffer != NULL);
+    assert(bufferSize > 0);
+
+    if (buffer == NULL || bufferSize <= 0)
+    {
+        errno = EINVAL;
+        return -1;
+    }
+
+    struct passwd u;
+    struct passwd* pu; 
+
+    errno = 0;
+    int e;
+    while ((e = getpwuid_r(uid, &u, buffer, bufferSize, &pu)) == EINTR);
```

Comment:
The buffer passed in isn't just used for the username. It's used for all strings referenced by `struct passwd`.
You need to access `pw_name` to get the name, otherwise you're assuming the name gets stored first in the buffer which is implementation-defined.